Index generation

tduberne opened the pull request:
This PR introduces two commands:
- ZettelGenerateIndex
- ZettelGenerateTags

They are based on their Vimwiki counterparts, the difference being that the links are named using the `%title` attribute instead of the file name. 

If there is interest in this PR, I could add a few more comments before merging.

michal-h21 commented:
Thanks, this is great!

michal-h21 commented:
Shouldn't be the `:ZettelGenerateIndex`command named as `:ZettelGenerateLinks`, like the corresponding Vimwiki command?

tduberne commented:
Renaming makes sense, I changed the name based on my own workflow and preferences only. I updated the branch on my fork and will make a new PR.

Thanks to you for the nice plugin!

michal-h21 commented:
Thank you. I've managed to add a new feature thanks to your code, backlinks. So this was really useful PR.
